x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/LucenePushdownPredicates.java

```diff
@@ -96,6 +104,46 @@ static String pushableAttributeName(TypedAttribute attribute) {
             : attribute.name();
     }
 
+    /**
+     * Extract the real field name from a MultiTypeEsField, limit to MultiTypeEsField that has date_nanos type only.
+     *
+     * For example, the name of a MultiTypeEsField can be $$myfield$converted_to$date_nanos, and the real field name extract from the
+     * MultiTypeEsField is myfield, this method return myfield given a MultiTypeEsField.
+     *
+     * If the real field name is found, and the original field data types contain only date and date_nanos types, return the real field
+     * name, so that the real field name will be used to check for eligibility of being pushed down, and the real field name will be used
+     * in the push down query, instead of the name of the MultiTypeEsField, which should not match any field in an index.
```

Comment:
The name of the MultiTypeEsField should still be the actual field name, btw - but we take liberty with the name of field attributes, these can be whatever these days.